tests/datatypes/test_tuple.py

```diff
@@ -180,6 +181,44 @@ def test_len(self):
         """)
 
 
+class MagicMethodFunctionTests(MagicMethodFunctionTestCase, TranspileTestCase):
+    data_type = 'tuple'
+    MagicMethodFunctionTestCase._add_tests(vars(), tuple)
```

Comment:
This "class setup" pattern is a little incongruous with the way the other operator transpile tests are configured - is there reason this approach is needed?

Reply:
It is a bit different. I'm trying to create different sets of tests depending on the properties of the python type being tested. If I create the tests at the parent class level like the other tests do, I don't have knowledge of type yet and can't select which tests are correct. I'll take another run at it and see if I can make it more straightforward to use.

Reply:
I'm not seeing an easy way to straighten it out. Do you think it is an issue to do it this way?

Reply:
Ah - I hadn't caught on that this was in aid of doing a different construction based on the type being added. There's possibly something nifty that could be done with metaclasses, but then we'd spend as much time fixing the metaclass as we'd do using the darn thing - so this "longhand" version seems fine to me.